src/ui.text-badge/UBadge.vue

```diff
@@ -1,3 +1,91 @@
+<script lang="ts" setup>
+import { ref, computed } from "vue";
+
+import { getDefault } from "../utilsTs/utilUI.ts";
+import UIcon from "../ui.image-icon/UIcon.vue";
+
+import { UBadge } from "./constants.ts";
+import useAttrs from "./useAttrs.ts";
+import defaultConfig from "./config.ts";
+
+import type { UBadgeProps } from "./types.ts";
+
+defineOptions({ inheritAttrs: false });
+
+const props = withDefaults(defineProps<UBadgeProps>(), {
+  variant: getDefault<UBadgeProps>(defaultConfig, UBadge).variant,
+  bordered: getDefault<UBadgeProps>(defaultConfig, UBadge).bordered,
+  size: getDefault<UBadgeProps>(defaultConfig, UBadge).size,
+  color: getDefault<UBadgeProps>(defaultConfig, UBadge).color,
+  round: getDefault<UBadgeProps>(defaultConfig, UBadge).round,
+  tabindex: getDefault<UBadgeProps>(defaultConfig, UBadge).tabindex,
+});
+
+const emit = defineEmits([
+  /**
+   * Triggers when the badge is focused.
+   */
+  "focus",
+
+  /**
+   * Triggers when the badge is pressed.
+   */
+  "keydown",
+
+  /**
+   * Triggers when the badge loses focus.
+   */
+  "blur",
+
+  /**
+   * Triggers when the badge is clicked.
+   */
+  "click",
+]);
+
+const { badgeAttrs, bodyAttrs, leftIconAttrs, centerIconAttrs, rightIconAttrs } = useAttrs(props);
+
+const wrapperRef = ref(null);
```

Comment:
Please use this approach: https://vuejs.org/guide/typescript/composition-api.html#typing-template-refs